dask/dataframe/core.py

```diff
@@ -540,9 +540,79 @@ def sample(self, frac, replace=False, random_state=None):
+        """ Export frame to hdf file(s)
+
+        Export dataframe to one or multiple hdf5 files or nodes.
+
+        Exporting dataframe to multiple targets enables improved parallelism
+        and is expected to provide better performance for longer computations.
+
+        Parameters
+        ----------
+        path_or_buf: HDFStore object or string
+            Destination file(s). If string, can contin a single wildcard
+            to save each partition to a different file. Only one widlcard
+            is allowed in both path and key,
+        key: string
+            A node / group identifier, can contain a single wildcard
+            to save each partition to a different hdf node in a single
+            file. This multiple better parallelism compared to saving
+            entire dataframe in a single node.
+        mode: optional, {'a', 'w', 'r+'}, default 'a'
+
+          ``'a'``
+              Append: Add data to existing file(s) or create new.
+          ``'w'``
+              Write: overwrite any existing files with new ones.
+          ``'r+'``
+              Append to existing files, files must already exist.
+        append: optional, default False
+            When False, overwrites existing node with the same name
+            otherwise appends to it.
+        complevel: optional, 0-9, default 0
+            compression level, higher means better compression ratio
+            and possibly more CPU time. depends on complib.
+        complib: {'zlib', 'bzip2', 'lzo', 'blosc', None}, default None
+            if complevel > 0 compress using this compression library
+            when possible
+        fletcher32: bool, default False
+            when True and compression is used, additionally apply the
+            fletcher32 checksum.
+        get: callable, optional
+            A scheduler `get` function to use. If not provided, the
+            default is to check the global settings first, and then
+            fall back to defaults for the collections.
+        dask_kwargs: dict, optional
+            A dictionary of keyword arguments passed to the `get` function
+            used.
+        name_function: callable, optional, default None
+            A callable called for each partition that accepts a single int
+            representing the partition number. name_function must return
+            a string representation of a partition's index in a way that
+            will preserve the partition's location after a string sort.
+
+            When None is received, a default name_function is used. The
+            default name_function will return a zero padded string of
+            received int. See dask.utils.build_name_function for more info.
+        compute: bool, default True
+            If True, execute computation of resulting dask graph.
+            If False, return a Delayed object.
+        lock: bool, None or lock object, default None
```

Comment:
This deserves some explanation about what locking here means, and why would someone use some locker or another.